core/src/main/java/tc/oc/pgm/map/MapInfoImpl.java

```diff
@@ -255,6 +266,19 @@ public Component getStyledName(MapNameStyle style) {
     return rules;
   }
 
+  private static List<Gamemode> parseGamemodes(Element root) throws InvalidXMLException {
+    List<Gamemode> gamemodes = new ArrayList<>();
+    for (Element gamemodeEl : root.getChildren("gamemode")) {
+      Gamemode gm = Gamemode.byId(gamemodeEl.getText());
+      if (gm == null) {
+        throw new InvalidXMLException("Unknown gamemode", gamemodeEl);
+      } else {
+        gamemodes.add(gm);
+      }
```

Comment:
```suggestion
      if (gm == null)
        throw new InvalidXMLException("Unknown gamemode", gamemodeEl);
    gamemodes.add(gm);
```
no need for the else, since the exception will already stop further execution